app/controllers/users_controller.rb

```diff
@@ -101,14 +107,9 @@ def create
       if current_user.invalid?
         # Something is wrong with a new user, so rerender the form
         render :action => "new"
-      elsif current_user.auth_provider.present?
-        # Verify external authenticator before moving on
-        session[:new_user] = current_user.slice("email", "display_name", "pass_crypt", "pass_crypt_confirmation")
```

Comment:
`session[:new_user]` filtered out other parameters such as `auth_provider` which was added later in `auth_success`. A new users was created from `session[:new_user]`.

With `session[:new_user]` gone its possible to create users with fake values for `auth_provider` by opening the signup form like this: `/user/new?auth_provider=test1&auth_uid=test1`

Reply:
I believe that user would not be created immediately after `/user/new`.

They would still have to click on "Sign up" on the screen that offers them to confirm their display name. From here, they would have to verify their email as well, unless they also correctly guess the value of `verified_email` value. Finally, they would probably have to reset their password as well, having no other way to login, and password being set up to a random value.

If this is a real concern, I can bring back the extra round trip to the external provider before creating user record.

Reply:
> > They would still have to click on "Sign up"
> 
> Yes, but right after that the user is written to the db with arbitrary auth_provider. That happens before any further confirmation.

Correct. This is pretty much same as if user is created by regular "signup" screen, before they confirm the email address. The only difference is that there is arbitrary, not usable, value of `auth_provider`, and the user does not know the password assigned to them.

I removed the extra round trip after @tomhuges suggestion here: https://github.com/openstreetmap/openstreetmap-website/pull/4455#discussion_r1443987257.

Having a user with bogus value of `auth_provider` does not hurt OSM web site. While the user cannot use the bogus value that they manually entered, they can use the account by username/password after resetting the password. Since there is a manual action required to actually create the user record, there is no danger of bulk creation of user accounts this way, more than it is possible today.

Reply:
@tomhughes I am planning to re-introduce the verification round trip, due to security concerns raised by @AntonKhorev

Reply:
What "security concern" is that? Why do we care if somebody creates a user with a "fake" authentication provider? What exactly is the claimed attack vector here?

Reply:
That user would never be able to login with the provider in question because it's not a provider we know anything about and if we do decide we care we can just validate that parameter against the expected values - no need to do a revalidation that I can see.

Reply:
@tomhughes - sorry, I didn't see your comment before pushing the extra round-trip. It is a separate commit so it would be easy to remove.

I don't really see the attack vector here, other than users creating accounts for which they would have to reset the password.

Reply:
@tomhughes , @AntonKhorev Can we reach consensus here?

I don't mind it either way, but I prefer the no-verification round trip purely due to being more responsive to user.

Reply:
User A gives user B a registration url with planted auth provider.

Reply:
Then what? What happens to a user with an auth provider of `fred` recorded in the database that causes a problem? Other than that they can't login until they do a password reset if they didn't chose a password when creating the account.